iamclient/iamclient.go

```diff
@@ -838,39 +839,55 @@ func (client *Client) GetPermissions(
 func (client *Client) openGRPCConnection() (err error) {
 	log.Debug("Connecting to IAM...")
 
-	var publicConn *grpc.ClientConn
+	var publicConn, protectedConn *grpc.ClientConn
 
 	publicConn, err = grpchelpers.CreatePublicConnection(
 		client.publicURL, client.cryptocontext, client.insecure)
 	if err != nil {
 		return aoserrors.Wrap(err)
 	}
 
-	if err := client.publicConnection.Start(publicConn); err != nil {
-		return aoserrors.Wrap(err)
-	}
+	client.publicConnection.Set(publicConn)
+
+	defer func() {
```

Comment:
Usually, no need to put good path into defer for readability reason.

```suggestion
	defer func() {
	     if err != nil {
	         publicConn.Close()
	     }
	 }
	 
	 ...
	 
	 client.publicConnection.Start()
```

Reply:
client.publicConnection.Start should be done in the end of the function execution to ensure connection succeded. Otherwise we unlock pending requests earlier & they can fail. 
Current function beside error check has also 
(client.isProtectedConnEnabled) logic, which means we should duplicate
client.publicConnection.Start(), to avoid that I decided to add both cases to defer() function.